syntax/org.vim

```diff
@@ -111,5 +111,22 @@ highlight def link orgHyperCentre Comment
 highlight def link orgHyperLeft Comment
 highlight def link orgHyperRight Comment
 
+" TeX
+" Support for both inline and block based embedded latex
+" eg: $Latex$ for inline or $$ LaTeX $$ for a block
+" Note:
+" - $LaTeX$ uses the tex.vim syntax for its conceal properties
+" - Inspired by https://github.com/vim-pandoc/vim-pandoc-syntax
+" - the conceal settings follows your g:tex_conceal setting for
+"   more info run :h tex-conceal.
+" Ref: https://orgmode.org/manual/LaTeX-fragments.html#LaTeX-fragments
+syntax include @LATEX syntax/tex.vim
+syntax region orgMath     start="\\begin[.*]{.*}"  end="\\end{.*}" 		 keepend contains=@LATEX
+syntax region orgMath     start="\\begin{.*}" 	 end="\\end{.*}" 		 keepend contains=@LATEX
+syntax region orgMath     start="\\\[" 				 end="\\\]" 			 keepend contains=@LATEX
```

Comment:
Looking at this reference you gave it seems there is another variant of this syntax:

```
$$ x^2 $$ and \[ x^2 \] and \( x^2 \) are all equivalent.
```

You've missed the 3rd one.  It can be implemented like so:

```suggestion
syntax region orgMath start="\\\[" end="\\\]" keepend contains=@LATEX
syntax region orgMath start="\\("  end="\\)"  keepend contains=@LATEX
```